src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javax.crypto;
+
+import java.security.*;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.InvalidParameterSpecException;
+import java.security.Provider.Service;
+import java.security.NoSuchAlgorithmException;
+import java.util.Iterator;
+import java.util.Objects;
+
+import sun.security.util.Debug;
+import sun.security.jca.*;
+import sun.security.jca.GetInstance.Instance;
+
+import javax.crypto.spec.KDFParameterSpec;
+
+/**
+ * This class provides the functionality of a key derivation algorithm for the Java Cryptographic
+ * Extension (JCE) framework.
+ * <p>
+ * {@code KeyDerivation} objects will be instantiated through the {@code getInstance} family of
+ * methods.  Key derivation algorithm names will follow a naming convention of
+ * <I>algorithm</I>/<I>PRF</I>.  The algorithm field will be the KDF name
+ * (e.g. HKDF, TLS-PRF, PBKDF2, etc.), while the PRF specifier identifies the underlying
+ * pseudorandom function (e.g. HmacSHA256).  For instance, a KDF implementation of HKDF using
+ * HMAC-SHA256 will have an algorithm string of "HKDF/HmacSHA256".  In some cases the PRF portion of
+ * the algorithm specifier may be omitted if the KDF algorithm has a fixed or default PRF.
+ * <p>
+ * TODO: finish this javadoc
+ */
+
+public final class KDF {
+    private static final Debug debug = Debug.getInstance("jca", "KeyDerivation");
```

Comment:
Change to a shorter option name "kdf". Also, add it into the help output of `sun.security.util.Debug`.